core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -60,11 +60,11 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   var name: String = null
   var childArgs: ArrayBuffer[String] = new ArrayBuffer[String]()
   var jars: String = null
-  var packages: String = null
-  var repositories: String = null
-  var ivyRepoPath: String = null
+  var packages: Option[String] = None
```

Comment:
Sorry but why do we bother change this? Looks like there are a lot of `null`s above too.

Reply:
> Sorry but why do we bother change this? Looks like there are a lot of `null`s above too.

Yea, that's why I asked  about  https://github.com/apache/spark/pull/30928#issuecomment-751213946

Since want to refactor parameter to use `Option[String]`, but here are just `String`, so change this place .

Reply:
`String` vs `null` is a valid way probably in Java context. `Option` might be preferred since we're in Scala but I wouldn't change it just for this reason. The gain is small compared to the size of change and potential maintenance overhead.

Reply:
> `String` vs `null` is a valid way probably in Java context. `Option` might be preferred since we're in Scala but I wouldn't change it just for this reason. The gain is small compared to the size of change and potential maintenance overhead.

So would you prefer to keep current or revert this change and just make it as Option when use method 
```
      val resolvedMavenCoordinates = DependencyUtils.resolveMavenDependencies(
        packagesTransitive = true, Option(args.packagesExclusions), Option(args.packages),
        Option(args.repositories), Option(args.ivyRepoPath), Option(args.ivySettingsPath)
```

Reply:
Yeah, I wouldn't particularly mind when I read these codes. If you still prefer, feel free to wait for other committers' approval or review. I don't mind if any committer prefers and wants to merge it in.

Reply:
Ok， wait for other committer's review.

Reply:
I'd be neutral on it - but if we're just changing a few instances? not sure it's worth the inconsistency.

Reply:
> I'd be neutral on it - but if we're just changing a few instances? not sure it's worth the inconsistency.

I decide to change this since there is already a Option[String]
```
var ivySettingsPath: Option[String] = None
```

Reply:
I'm neutral on this change, too. I find `xxx: String = null` in many places of the `core` package and I'm not sure that this partial fix can make any benefit.

Reply:
> I'm neutral on this change, too. I find `xxx: String = null` in many places of the `core` package and I'm not sure that this partial fix can make any benefit.

Ok, revert change. about this.How about current